consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -614,45 +791,267 @@ impl PruningProofManager {
 
         drop(pruning_read);
         drop(relations_read);
-        drop(db_lifetime);
+        drop(proof_stores_and_processes.db_lifetime);
+        drop(current_consensus_stores_and_processes.db_lifetime);
 
         Err(PruningImportError::PruningProofNotEnoughHeaders)
     }
 
+    // The "current dag level" is the level right before the level whose parents are
+    // not the same as our header's direct parents
+    //
+    // Find the current DAG level by going through all the parents at each level,
+    // starting from the bottom level and see which is the first level that has
+    // parents that are NOT our current pp_header's direct parents.
+    fn find_current_dag_level(&self, pp_header: &Header) -> BlockLevel {
+        let direct_parents = BlockHashSet::from_iter(pp_header.direct_parents().iter().copied());
+        pp_header
+            .parents_by_level
+            .iter()
+            .enumerate()
+            .skip(1)
+            .find_map(|(level, parents)| {
+                if BlockHashSet::from_iter(parents.iter().copied()) == direct_parents {
+                    None
+                } else {
+                    Some((level - 1) as BlockLevel)
+                }
+            })
+            .unwrap_or(self.max_block_level)
+    }
+
+    fn estimated_blue_depth_at_level_0(&self, level: BlockLevel, level_depth: u64, current_dag_level: BlockLevel) -> u64 {
+        level_depth.checked_shl(level.saturating_sub(current_dag_level) as u32).unwrap_or(level_depth)
+    }
+
+    /// selected parent at level = the parent of the header at the level
+    /// with the highest blue_work (using score as work in this case)
```

Comment:
~~(using score as work in this case)~~ 

we're actually using the header blue work here

Reply:
the text `(using score as work in this case)` still needs to be removed